### PR TITLE
feat: added reorderable list and updated fieldProps

### PR DESCRIPTION
**Pull request by Tormak9970**
Added the reorderable list component and added `style` property to the `FieldProps` type.

Example of how to implement the reorderable list can be found [here](https://github.com/Tormak9970/bash-shortcuts/blob/3be7c232c64158636dfb8f0f4550086def873394/src/components/shortcuts-manager/ManageShortcuts.tsx#L14)

### CODE REVIEW

File: src/custom-components/ReorderableList.tsx [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    return (
+        <Fragment>
+            <style>{`
+                .scoper {
+                    width: 100%;
+                    display: flex;
+                    flex-direction: column;
+                }
+            `}</style>
+            <div className="scoper"
```
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:268]: Class name is a bit too generic. Also why not just use an inline style here.

File: src/custom-components/ReorderableList.tsx [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                    focusIdx.current++;
+                } else {
+                    focusIdx.current--;
+                }
+            }
+        }
+
+        return (
+            <Fragment>
+                <div className="custom-buttons">
```
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:112]: Class name too generic

**Review by AAGaming00 — DISMISSED** [review]

File: src/custom-components/ReorderableList.tsx [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    
+    useEffect(() => {
+        dataAsList = [];
+        dataAsList = Object.values(props.data).sort((a, b) => a.data.position - b.data.position);
+        data = props.data;
+    });
+
+    function enableReorder() { reorderEnabled.current = true; }
+    function disabledReorder() { reorderEnabled.current = false; }
+    function forceUpdate() { setUpdate(update === 0 ? 1 : 0); }
```
[the remark this answers is not in the document]
Comment (Tormak9970) [reply src/custom-components/ReorderableList.tsx:63]: From what I understand that is the intended use of useref. Also why convert the to arrow functions?

I'll rename the classes and inline the scoper style when I have a chance.

**Comment by Tormak9970** [comment]
Working on a new Toggle version that's simpler and more useful. Will update the PR when its done and tested.

**Comment by Tormak9970** [comment]
@AAGaming00 Would appreciate you taking a look in case you want me to make changes.

File: src/custom-components/ReorderableList.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+          flex: 1 1 1px;
+          scroll-padding: 48px 0px;
+          display: flex;
+          flex-direction: column;
+          justify-content: flex-start;
+          align-content: stretch;
+        }
+
+        .reorderable-list .custom-buttons {
```
Comment (EMERALD0874) [inline src/custom-components/ReorderableList.tsx:50]: Is this style necessary?

File: src/custom-components/ReorderableList.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  );
+}
+
+type ListEntryProps<T> = {
+  fieldProps?: FieldProps,
+  listData: ReorderableEntry<T>[],
+  entryData: ReorderableEntry<T>,
+  reorderEntryFunc: CallableFunction,
+  reorderEnabled: boolean,
+  children:ReactElement|null
```
Comment (EMERALD0874) [inline src/custom-components/ReorderableList.tsx:80]: You may want to have a Decky maintainer run this through a linter so it matches our Prettier config.

File: package.json [hunk 6]
```diff
@@ -78,5 +78,8 @@
       "style": "module",
       "parser": "typescript"
     }
+  },
+  "dependencies": {
+    "react-icons": "^4.6.0"
```
Comment (EMERALD0874) [inline package.json:83]: react-icons is not used in this PR. Please revert changes to package.json and pnpm-lock.yaml as this was the only change made to both.

**Review by EMERALD0874 — CHANGES_REQUESTED** [review]
Please see the attached comments.

**Review by EMERALD0874 — APPROVED** [review]
Looks good! Just please make sure a Decky maintainer runs this through Prettier before merge.

File: src/custom-components/ReorderableList.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+  entries: ReorderableEntry<T>[],
+  onSave: (entries: ReorderableEntry<T>[]) => void,
+  interactables?: JSXElementConstructor<{entry:ReorderableEntry<T>}>,
+  fieldProps?: FieldProps
+}
+
+/**
+ * A component for creating reorderable lists.
+ * 
+ * Implementation example can be found {@link https://github.com/Tormak9970/Component-Testing-Plugin/blob/main/src/testing-window/ReorderableListTest.tsx here}.
```
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:20]: use `@example` for better doc

File: src/custom-components/ReorderableList.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
+import { Fragment, JSXElementConstructor, ReactElement, useEffect, useState } from "react";
+import { Field, FieldProps, Focusable, GamepadButton } from "../deck-components";
+
+export type ReorderableEntry<T> = {
```
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:4]: add documentation for these

File: src/custom-components/ReorderableList.tsx [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+              </ReorderableItem>
+            ))
+          }
+        </Focusable>
+      </div>
+    </Fragment>
+  );
+}
+
+type ListEntryProps<T> = {
```
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:72]: see first comment

File: src/custom-components/ReorderableList.tsx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+    setReorderEnabled(newReorderValue);
+
+    if (!newReorderValue){
+      props.onSave(entryList);
+    }
+  }
+
+  return (
+    <Fragment>
+      <style>{`
```
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:41]: inline style probably better here, we don't want dfl to mess with global CSS too much

File: src/custom-components/ReorderableList.tsx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+  const baseCssProps = {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "space-between",
+    width: "100%"
+  };
+
+  return(
+    // @ts-ignore
```
Comment (AAGaming00) [inline src/custom-components/ReorderableList.tsx:120]: why the `@ts-ignore`?

**Review by AAGaming00 — DISMISSED** [review]

**Comment by AAGaming00** [comment]
> Looks good! Just please make sure a Decky maintainer runs this through Prettier before merge.

Seems like maintainer access isn't working as Tormack didn't create a new branch on their fork for this PR.

```
❯ git push
remote: Permission to Tormak9970/decky-frontend-lib.git denied to AAGaming00.
fatal: unable to access 'https://github.com/Tormak9970/decky-frontend-lib/': The requested URL returned error: 403
```

**Comment by Tormak9970** [comment]
Will make the requested changes and run prettier myself

File: src/custom-components/ReorderableList.tsx (hunk 7 above)
Comment (Tormak9970) [reply src/custom-components/ReorderableList.tsx:20]: You want me to put the example in the jsDoc comment?

File: src/custom-components/ReorderableList.tsx (hunk 11 above)
Comment (Tormak9970) [reply src/custom-components/ReorderableList.tsx:120]: fixed by specifying type on css obj

File: src/custom-components/ReorderableList.tsx (hunk 5 above)
Comment (Tormak9970) [reply src/custom-components/ReorderableList.tsx]: ran prettier on newest update

**Review by TrainDoctor — APPROVED** [review]

**Review by AAGaming00 — APPROVED** [review]

**Pull request merged**